client/post-editor/editor-ground-control/index.jsx

```diff
@@ -123,8 +123,10 @@ export default React.createClass( {
 	},
 
 	setPostDate: function( date ) {
+		const dateValue = date ? date.format() : null;
 		// TODO: REDUX - remove flux actions when whole post-editor is reduxified
-		postActions.edit( { date: date ? date.format() : null } );
+		postActions.edit( { date: dateValue } );
+		this.showDateWarning( dateValue );
```

Comment:
IMHO it should just be `checkForDateChange()` and that method should call the methods that actually do things. Maybe we'll want to do other things in the future based on a date change.